go/vt/workflow/node.go

```diff
@@ -0,0 +1,355 @@
+package workflow
+
+import (
+	"encoding/json"
+	"fmt"
+	"path"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/golang/glog"
+	"golang.org/x/net/context"
+
+	"github.com/youtube/vitess/go/vt/topo"
+
+	workflowpb "github.com/youtube/vitess/go/vt/proto/workflow"
+)
+
+// This file contains the necessary object definitions and interfaces
+// to expose a workflow display to the vtctld webapp.
+//
+// Most constants and variable names should match the Node object described in
+// web/vtctld2/src/app/workflows/node.ts as it is exposed as JSON to
+// the angular2 web app.
+
+// NodeDisplay constants need to match node.ts.Display.
+type NodeDisplay int
+
+const (
+	// NodeDisplayIndeterminate is a progress bar that doesn't have
+	// a current value, but just shows movement.
+	NodeDisplayIndeterminate NodeDisplay = 0
```

Comment:
I started with iota, but I like the explicit numbers better. (and honestly I couldn't get iota to work with typed constants... :) ).